assets/kube-state-metrics/deployment.yaml

```diff
@@ -35,7 +35,10 @@ spec:
         - --port=8081
         - --telemetry-host=127.0.0.1
         - --telemetry-port=8082
-        - --metric-denylist=kube_secret_labels,kube_.*_annotations
+        - |
+          --metric-denylist=
+          ^kube_secret_labels$,
+          ^kube_.+_annotations$
```

Comment:
doesn't this create  `"--metric-denylist=\n^kube_secret_labels$,\n^kube_.+_annotations$\n"` ?

Reply:
yes, it does but KSM seem to be interpreting correctly, The same pattern has been used in [upstream mixins too](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/addons/ksm-lite.libsonnet#L18).

Also I had verified these metrics are not emitted by KSM.

Reply:
I see that is how it is applied upstream  but I wonder if `addArgs` fixes the `\n` .. 
Are you able to add a test to https://github.com/kubernetes/kube-state-metrics/blob/ef8bede4064f04472a8c2fe8c281e6bf97de4329/pkg/options/types_test.go#L165 to see if that really works? my experience says otherwise